snuba/datasets/schemas/join.py

```diff
@@ -70,6 +84,16 @@ class SubJoinSource(JoinedSource):
     def __str__(self) -> str:
         return f"{self.structure.get_data_source()}"
 
+    def get_schemas(self) -> Mapping[str, Schema]:
+        left = self.structure.left_source.get_schemas()
+        right = self.structure.right_source.get_schemas()
+        overlapping_aliases = left.keys() & right.keys()
+        for alias in overlapping_aliases:
+            # Ensures none defines the same alias twice in the join referring
+            # to different tables.
+            assert left[alias] == right[alias]
```

Comment:
What's the case where you'd expect these to be equal?

Reply:
(just an example, not a real join)

> events A LEFT JOIN groups B ON A.group_id = B.id LEFT JOIN group_assignee C ON A.group_id = C.group_id

Depending on how this is represented, the alias A referring to events will be present twice in the join clause

Reply:
Ahh ok, this makes sense, thanks.